### ISSUE TITLE
Systemd doesn't use /dev/urandom during boot to create an encrypted device

### ISSUE THREAD

**Issue by StayPirate** [systemd/systemd#11360]
**systemd version the issue has been seen with**
> 239.370-1

**Used distribution**
 > Archlinux

**Expected behaviour you didn't see**
 > Creating an encrypted swap device using a random key

**Unexpected behaviour you saw**
 > I'm being prompt to input a passphrase for the encrypted swap device while I set it to use /dev/urandom as key

**Steps to reproduce the problem**

Configuration files:

**/etc/crypttab**
```swap            UUID=94a7a590-2ac3-42f2-8ba6-1533fca05638       /dev/urandom    swap,offset=2048,plain,cipher=aes-xts-plain64,size=256```

**Kernel cmdline**
```console=ttyS0,115200 luks.name=593cbd9a-be3b-47b7-a79e-509e867e5ba5=Arch luks.uuid=94a7a590-2ac3-42f2-8ba6-1533fca05638 ip=:::::eth0:dhcp```

**/etc/fstab**
```/dev/mapper/swap        none    swap    defaults        0 0```

**UUID**
```94a7a590-2ac3-42f2-8ba6-1533fca05638``` is my **swap** partition
```593cbd9a-be3b-47b7-a79e-509e867e5ba5``` is my **root** partition

**Boot logs**
```
Starting Cryptography Setup for luk…590-2ac3-42f2-8ba6-1533fca05638...
Starting Cryptography Setup for Arch...
Please enter passphrase for disk Linux\x20dm-crypt (Arch): *************************
Please enter passphrase for disk Linux\x20swap (luks-94a7a590-2ac3-42f2-8ba6-1533fca05638): 
[FAILED] Failed to start Cryptography Setup …7a590-2ac3-42f2-8ba6-1533fca05638.
See 'systemctl status "systemd-cryptset…\x2d1533fca05638.service"' for details.
[DEPEND] Dependency failed for Local Encrypted Volumes.
[  OK  ] Reached target System Initialization.
[  OK  ] Reached target Basic System.
[  OK  ] Started Cryptography Setup for Arch.
```
Following instruction on logs ```See 'systemctl status "systemd-cryptset…\x2d1533fca05638.service"' for details.```, this service doesn't exist anymore after the system completes the boot phase. I cannot figure out why I'm being prompted to input a passphrase for the swap device, even if I input an empty string the system will continue to proper boot and actually creates the right device and mount it as swap (as you can see from the following command output):
```
NAME FSTYPE LABEL UUID                                 FSAVAIL FSUSE% MOUNTPOINT
sda
├─sda1
│
├─sda2
│    ext4   Boot  6cac7d99-d884-4f42-96b8-8f7839598376  156.9M    26% /boot
├─sda3
│    crypto       593cbd9a-be3b-47b7-a79e-509e867e5ba5
│ └─Arch
│    btrfs  Root  27838434-df75-45f4-b6dd-daad7fa16dca  459.4G     0% /
└─sda4
     ext2   cryptswap
                  94a7a590-2ac3-42f2-8ba6-1533fca05638
  └─swap
     swap         14a74f58-504c-4ddb-9820-be8a57b98a64                [SWAP]
```

What should I do to let systemd use /dev/urandom as a key for the swap device during boot instead of asking me to input a passphrase?

**Comment by poettering** [comment]
Does it work if you specify systemd.gpt_auto=no on the kernel cmdline?

**Comment by poettering** [comment]
maybe your shell at up the `\` in that cmdline?

but yeah, debug logs of the relevant bits are of course always helpful?

**Comment by mejo-** [comment]
I experience the same bug in a non-deterministical way. One of my VMs has a crypttab entry with '/dev/urandom' as keyfile and option 'tmp'. When I reboot the VM, sometimes it boots up and the dm-crypt device is unlocked as expected, sometimes it gets stuck at a passphrase prompt for the dm-crypt device in the boot process. I don't change anything on the system in between the reboots. I didn't find a way to reprodce the bug deterministically yet.

**Comment by poettering** [comment]
the requested debug logs where never provided...

I am pretty sure this is caused by the same issue as #10179 which is now fixed in #15836. If it is still reproducible with a version that includes that PR, please open a new issue.
